ujt/transformers.py

```diff
@@ -373,3 +373,44 @@ def apply_slis_to_node_map(sli_list, node_map):
         node.slis.extend(new_node_slis)
 
     return node_map
+
+
+def sort_nodes_by_parent_relationship(elements):
+    """Returns a list of elements where node parents always appear before node children.
+
+    For context, see https://github.com/plotly/dash-cytoscape/issues/112
+    and https://github.com/googleinterns/userjourneytool/issues/63
+
+    Args:
+        elements: a list of cytoscape elements.
+
+    Returns:
+        a list of topologically sorted cytoscape elements.
+    """
+    edges = []
+    node_id_element_map = {}
+    for element in elements:
+        if utils.is_node_element(element):
+            node_id_element_map[element["data"]["id"]] = element
+        else:
+            edges.append(element)
+
+    parent_child_map = defaultdict(list)
+    bfs_queue = deque()
+    # build a tree from parents to children
+    # (reversing the edge direction of cytoscape format)
+    for node in node_id_element_map.values():
+        node_id = node["data"]["id"]
+        if "parent" in node["data"]:
+            parent_id = node["data"]["parent"]
+            parent_child_map[parent_id].append(node_id)
+        else:
+            bfs_queue.append(node_id)
```

Comment:
This could be all be done in the first pass through elements, but I split the loops up to make the operations more clear to readers.